### ISSUE TITLE
Hardware keyboard required?

### ISSUE THREAD

**Issue by hackerb9** [soupslurpr/BeauTyXT#161]
The official website doesn't mention anything about this, but I'm wondering if this app is designed for use with an external keyboard because I keep running into issues without one.

The biggest problem is that when the Markdown preview window is being shown, it cannot be scrolled correctly if an on screen keyboard is in use.

Here are screenshots showing the problem. 

1. When at the top of a file, the preview is correct.
![1000021087](https://github.com/user-attachments/assets/8e91238c-2ab9-4688-9df2-94ece12cdc27)

2. Going to the bottom of the document, the preview does not scroll automatically
![1000021088](https://github.com/user-attachments/assets/0d558421-93f0-4140-ba96-c4c63934744f)

3. Scrolling the preview manually removes the on screen keyboard
![1000021089](https://github.com/user-attachments/assets/4c7907e9-d374-42a1-9c8b-9e4366f3d9a8)

4. Tapping in the edit window to edit the bottom of the file brings up the on screen keyboard,. Bug: neither the edit window nor the preview are showing the bottom of the file where the text cursor is at.
![1000021090](https://github.com/user-attachments/assets/5a334aa2-2656-4094-92b8-754c71574cf3)

5. Typing blindly into the edit window scrolls it so the bottom of the file is visible, but the preview window still shows the wrong location
![1000021091](https://github.com/user-attachments/assets/e577d475-f776-4a8c-bcba-bc3410537462)

**Comment by soupslurpr** [comment]
A hardware keyboard isn't required, but perhaps the experience is better with one. This isn't intentional though.

Automatically scrolling the preview to wherever the cursor is is a feature that's not implemented.

I've started a rewrite of BeauTyXT, I'll see about reworking this in a better way.

**Comment by hackerb9** [comment]
Thank you. Do you still want bug reports for the current version or should I hold off until after the rewrite? For example, when viewing a log file (1.5 MB), Beauty XT is very slow to scroll and repeatedly shows the "App Not Responding" error.

**Comment by soupslurpr** [comment]
Yes, please report any bugs you find now so I can make sure to not include them in the rewrite. Yeah, BeauTyXT struggles with large files due to Jetpack Compose's TextField not handling them well. I'm not sure how to solve that other than using a page based file format or using something else as the TextField that's more efficient.

**Comment by soupslurpr** [comment]
> [!NOTE]
> GPT-6-Astra, responding on behalf of soupslurpr.

[BeauTyXT Reimagined (version 84)](https://github.com/soupslurpr/BeauTyXT/releases/tag/84) is now released. It requires Android 17 / API 37 or newer.

The old simultaneous editor/preview layout was replaced with native rendered reading and separate source editing. Switching to preview hides the on-screen keyboard, and source/preview navigation preserves the document position. A hardware keyboard is not required. Closing the report about the old layout as addressed by the rewrite; please comment if a keyboard/scrolling problem remains in version 84.
